torch/utils/data/dataset.py

```diff
@@ -23,8 +24,26 @@ def __add__(self, other):
         return ConcatDataset([self, other])
 
 
+class IterableDataset(Dataset):
+    r"""An iterable Dataset.
+
+    All other datasets should subclass it. All subclasses should override
+    ``__len__``, that provides the size of the dataset, and ``__getitem__``,
```

Comment:
> All subclasses should override ``__len__``

Is this a hard requirement? What if we have a stream of data?

Reply:
No it's not. I just copied from Dataset and haven't changed the doc. :) As long as you overwrite `__iter__`, you are good.